README.md
# taco-se-tweaks

taco-se-tweaks is a mod that adds some personal QoL and items to work with the Space Exploration modpack.
This is only tested with Space Exploration V.0.6 (multiplayer).
Should work on existing saves.

# Features

<p>Changed landfill stacksize to 400 (may add a startup setting to let the user edit this at startup)</p>

<b>New items:</b>

- Tacoport mk3 (Personal roboport)
  - 20 charging slots
  - 50 robot limit
  - 10MW charging energy
  - Unlocked after energy science-3 and material science-3
- Taco exoskeleton
  - 50% movement bonus at 600kW consuption
  - Unlocked after energy science-3, material science-3 and bio science
- Doble decker wagon
  - cargo-wagon with 80 inventory slots
  - 1300 weight
  - 8000 health
- Taco fuel (train)
  - 2500% acceleration and 600% top speed

<b>New bonus reserach:</b>

- multiple trash slots (30 total) +infinity reserach with all science pack-4
- mulitple train braking force(300% total) +infinity reserach with all science pack-4
- multiple faster worker robotspeed(210%)
- mining productivity(20%)


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.1.1
Date: 08. 10. 2023
  Features:
    - added OP tacofuel with 2500% acceleration and 600% top speed (require all se-science pack 4 to reserach)
    - infinity research for autotrash and train breaking speed
  changes:
    - reduced Freddie's dobble decker wagon weight to 1300
    - increased all wagon's max_speed = 1000
  Bugfixes:
    - freiddie's doble decker wagon recipe is no longer enabled by defualt 
---------------------------------------------------------------------------------------------------
Version: 0.1.0
Date: 20. 09. 2023
  Major Features:
    - Added Taco exoskeleton
    - Two more train brake research
    - Added a doble decker wagon with doble inventory size and more health
    - Recipe cost changes
    - Added research to unlock new recipes
    - Added green to all new items/entities
---------------------------------------------------------------------------------------------------
Version: 0.0.3
Date: 17. 09. 2023
  Major Features:
    - Added  technology's for worker robot speed.
    - reduced technology cost.
---------------------------------------------------------------------------------------------------
Version: 0.0.2
Date: 16.09.2023
  Major Features:
    - Added 5 technology's for more trash slots.
    - Added 6 technology's for faster train braking speed.
---------------------------------------------------------------------------------------------------
Version: 0.0.1
Date: 15. 09. 2023
  Major Features:
    - Initial release.
    - Added a faster roboport and technology to unlock it.
    - Changed landfill stacksize to 400

prototypes/entities.lua
data:extend(
{
  {
    type = "cargo-wagon",
    name = "doble-decker-wagon",
    icon = "__taco-se-tweaks__/graphics/cargo-wagon.png",
    icon_size = 32,
    flags = {"placeable-neutral", "player-creation", "placeable-off-grid"},
    inventory_size = 80, --tweaked
    minable = {mining_time = 1, result = "doble-decker-wagon"},
    mined_sound = {filename = "__core__/sound/deconstruct-medium.ogg"},
    max_health = 8000, --tweaked
    corpse = "medium-remnants",
    dying_explosion = "medium-explosion",
    collision_box = {{-0.6, -2.4}, {0.6, 2.4}},
    selection_box = {{-1, -2.703125}, {1, 3.296875}},
    vertical_selection_shift = -0.796875,
    weight = 1300, --tweaked
    max_speed = 1.2,
    braking_force = 3,
    friction_force = 0.50,
    air_resistance = 0.01,
    connection_distance = 3,
    joint_distance = 4,
    energy_per_hit_point = 5,
    resistances =
    {
      {
        type = "fire",
        decrease = 15,
        percent = 50
      },
      {
        type = "physical",
        decrease = 15,
        percent = 30
      },
      {
        type = "impact",
        decrease = 50,
        percent = 60
      },
      {
        type = "explosion",
        decrease = 15,
        percent = 30
      },
      {
        type = "acid",
        decrease = 10,
        percent = 20
      }
    },
    back_light = rolling_stock_back_light(),
    stand_by_light = rolling_stock_stand_by_light(),
    color = {r = 0, g = 0, b = 0, a = 0},
    pictures =
    {
      layers =
      {
        {
          priority = "very-low",
          slice = 4,
          width = 222,
          height = 205,
          back_equals_front = true,
          direction_count = 128,
          allow_low_quality_rotation = true,
          filenames =
          {
            "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-1.png",
            "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-2.png",
            "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-3.png",
            "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-4.png"
          },
          line_length = 4,
          lines_per_file = 8,
          shift = {0, -0.796875},
          hr_version =
          {
            priority = "very-low",
            slice = 4,
            width = 442,
            height = 407,
            back_equals_front = true,
            direction_count = 128,
            allow_low_quality_rotation = true,
            filenames =
            {
              "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-1.png",
              "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-2.png",
              "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-3.png",
              "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-4.png"
            },
            line_length = 4,
            lines_per_file = 8,
            shift = util.by_pixel(0, -25.25),
            scale = 0.5
          }
        },
        {
          flags = { "mask" },
          priority = "very-low",
          slice = 4,
          width = 196,
          height = 174,
          direction_count = 128,
          allow_low_quality_rotation = true,
          back_equals_front = true,
          apply_runtime_tint = true,
          shift = {0, -1.125},
          filenames =
          {
            "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-mask-1.png",
            "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-mask-2.png",
            "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-mask-3.png"
          },
          line_length = 4,
          lines_per_file = 11,
          hr_version =
          {
            flags = { "mask" },
            priority = "very-low",
            slice = 4,
            width = 406,
            height = 371,
            direction_count = 128,
            allow_low_quality_rotation = true,
            back_equals_front = true,
            apply_runtime_tint = true,
            shift = util.by_pixel(-0.5, -30.25),
            filenames =
            {
              "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-mask-1.png",
              "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-mask-2.png",
              "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-mask-3.png"
            },
            line_length = 4,
            lines_per_file = 11,
            scale = 0.5
          }
        },
        {
          flags = { "shadow" },
          priority = "very-low",
          slice = 4,
          width = 246,
          height = 201,
          back_equals_front = true,
          draw_as_shadow = true,
          direction_count = 128,
          allow_low_quality_rotation = true,
          filenames =
          {
            "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-shadow-1.png",
            "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-shadow-2.png",
            "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-shadow-3.png",
            "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-shadow-4.png"
          },
          line_length = 4,
          lines_per_file = 8,
          shift = {0.8, -0.078125},
          hr_version =
          {
            flags = { "shadow" },
            priority = "very-low",
            slice = 4,
            width = 490,
            height = 401,
            back_equals_front = true,
            draw_as_shadow = true,
            direction_count = 128,
            allow_low_quality_rotation = true,
            filenames =
            {
              "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-shadow-1.png",
              "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-shadow-2.png",
              "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-shadow-3.png",
              "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-shadow-4.png"
            },
            line_length = 4,
            lines_per_file = 8,
            shift = util.by_pixel(32, -2.25),
            scale = 0.5
          }
        }
      }
    },
    horizontal_doors =
    {
      layers =
      {
        {
          filename = "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-door-horizontal-end.png",
          line_length = 1,
          width = 220,
          height = 33,
          frame_count = 8,
          shift = {0, -0.921875},
          hr_version =
          {
            filename = "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-door-horizontal-end.png",
            line_length = 1,
            width = 438,
            height = 63,
            frame_count = 8,
            shift = util.by_pixel(0, -29.25),
            scale = 0.5
          }
        },
        {
          filename = "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-door-horizontal-side.png",
          line_length = 1,
          width = 186,
          height = 38,
          frame_count = 8,
          shift = {0, -0.78125},
          hr_version =
          {
            filename = "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-door-horizontal-side.png",
            line_length = 1,
            width = 368,
            height = 76,
            frame_count = 8,
            shift = util.by_pixel(0, -24.5),
            scale = 0.5
          }
        },
        {
          filename = "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-door-horizontal-side-mask.png",
          width = 182,
          height = 35,
          line_length = 1,
          frame_count = 8,
          shift = {0, -0.828125},
          apply_runtime_tint = true,
          hr_version =
          {
            filename = "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-door-horizontal-side-mask.png",
            width = 320,
            height = 69,
            line_length = 1,
            frame_count = 8,
            shift = util.by_pixel(0, -26.25),
            apply_runtime_tint = true,
            scale = 0.5
          }
        },
        {
          filename = "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-door-horizontal-top.png",
          line_length = 1,
          width = 184,
          height = 28,
          frame_count = 8,
          shift = {0.015625, -1.125},
          hr_version =
          {
            filename = "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-door-horizontal-top.png",
            line_length = 1,
            width = 369,
            height = 54,
            frame_count = 8,
            shift = util.by_pixel(0.75, -35.5),
            scale = 0.5
          }
        },
        {
          filename = "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-door-horizontal-top-mask.png",
          width = 185,
          height = 23,
          frame_count = 8,
          line_length = 1,
          shift = {0.015625, -1.17188},
          apply_runtime_tint = true,
          hr_version =
          {
            filename = "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-door-horizontal-top-mask.png",
            width = 369,
            height = 45,
            frame_count = 8,
            line_length = 1,
            shift = util.by_pixel(0.75, -37.75),
            apply_runtime_tint = true,
            scale = 0.5
          }
        }
      }
    },
    vertical_doors =
    {
      layers =
      {
        {
          filename = "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-door-vertical-end.png",
          line_length = 8,
          width = 30,
          height = 23,
          frame_count = 8,
          shift = util.by_pixel(0, 62.5),
          hr_version =
          {
            filename = "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-door-vertical-end.png",
            line_length = 8,
            width = 58,
            height = 44,
            frame_count = 8,
            shift = util.by_pixel(0, 62.5),-- 241),--62.5+178.5),
            scale = 0.5
          }
        },
        {
          filename = "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-door-vertical-side.png",
          line_length = 8,
          width = 67,
          height = 169,
          frame_count = 8,
          shift = {0.015625, -1.01563},
          hr_version =
          {
            filename = "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-door-vertical-side.png",
            line_length = 8,
            width = 127,
            height = 337,
            frame_count = 8,
            shift = util.by_pixel(0.25, -32.75),
            scale = 0.5
          }
        },
        {
          filename = "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-door-vertical-side-mask.png",
          line_length = 8,
          width = 56,
          height = 163,
          frame_count = 8,
          shift = {0, -1.10938},
          apply_runtime_tint = true,
          hr_version =
          {
            filename = "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-door-vertical-side-mask.png",
            line_length = 8,
            width = 112,
            height = 326,
            frame_count = 8,
            shift = util.by_pixel(0, -35.5),
            apply_runtime_tint = true,
            scale = 0.5
          }
        },
        {
          filename = "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-door-vertical-top.png",
          line_length = 8,
          width = 32,
          height = 168,
          frame_count = 8,
          shift = {0, -1.125},
          hr_version =
          {
            filename = "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-door-vertical-top.png",
            line_length = 8,
            width = 64,
            height = 337,
            frame_count = 8,
            shift = util.by_pixel(0, -35.75),
            scale = 0.5
          }
        },
        {
          filename = "__taco-se-tweaks__/graphics/cargo-wagon/cargo-wagon-door-vertical-top-mask.png",
          line_length = 8,
          width = 32,
          height = 166,
          frame_count = 8,
          shift = {0, -1.15625},
          apply_runtime_tint = true,
          hr_version =
          {
            filename = "__taco-se-tweaks__/graphics/cargo-wagon/hr-cargo-wagon-door-vertical-top-mask.png",
            line_length = 8,
            width = 64,
            height = 332,
            frame_count = 8,
            shift = util.by_pixel(0, -37),
            apply_runtime_tint = true,
            scale = 0.5
          }
        }
      }
    },
    wheels = standard_train_wheels,
    rail_category = "regular",
    drive_over_tie_trigger = drive_over_tie(),
    tie_distance = 50,
    working_sound =
    {
      sound =
      {
        filename = "__base__/sound/train-wheels.ogg",
        volume = 0.6
      },
      match_volume_to_activity = true,
    },
    crash_trigger = crash_trigger(),
    open_sound = { filename = "__base__/sound/machine-open.ogg", volume = 0.85 },
    close_sound = { filename = "__base__/sound/machine-close.ogg", volume = 0.75 },
    sound_minimum_speed = 0.5;
    vehicle_impact_sound =  { filename = "__base__/sound/car-metal-impact.ogg", volume = 0.65 }
  },})
